components/admin_console/team_channel_settings/channel/details/channel_details.jsx

```diff
@@ -76,11 +78,19 @@ export default class ChannelDetails extends React.Component {
             then(() => this.setState({groups: this.props.groups}));
     }
 
-    setToggles = (isSynced, isPublic) => {
+    setToggles = (isSynced, isPublic, syncToggle) => {
```

Comment:
The new syncToggle is hard to understand in this context, I think you are putting this in the wrong level of abstraction, probably this should be in other component, maybe `ChannelModes`.

Reply:
That parameter was added to know when the sync toggle was used so I could restore the channel privacy toggle to the original state in case the sync toggle was left in the "off" state. Otherwise the channel privacy toggle would end up stuck in private state.

![Image](https://user-images.githubusercontent.com/1832946/64164381-71452b80-ce43-11e9-923b-9cab3cd3fc6a.gif)

Reply:
I could probably get rid of that parameter by keeping track of the sync toggle state in `ChannelModes`. I'll review this.

Reply:
I'll try to handle it inside `ChannelModes` which problably simplify things.